packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -172,60 +268,34 @@ export async function doDeployArmTemplates(ctx: SolutionContext): Promise<Result
-    setTimeout(async () => {
-      if (!deploymentFinished) {
-        const deployments = await client.deployments.listByResourceGroup(resourceGroupName);
-        deployments.forEach(async (deployment) => {
-          if (
-            deployment.properties?.timestamp &&
-            deployment.properties.timestamp.getTime() > deploymentStartTime
-          ) {
-            ctx.logProvider?.info(
-              `[${PluginDisplayName.Solution}] ${deployment.name} -> ${deployment.properties.provisioningState}`
-            );
-            if (deployment.properties.error) {
-              ctx.logProvider?.error(
-                `[${PluginDisplayName.Solution}] ${deployment.name} -> ${JSON.stringify(
-                  deployment.properties.error,
-                  undefined,
-                  2
-                )}`
-              );
-              const operations = await client.deploymentOperations.list(
-                resourceGroupName,
-                deploymentName
-              );
-              operations.forEach((op) => {
-                if (op.properties?.statusCode != "OK") {
-                  ctx.logProvider?.error(
-                    `[${PluginDisplayName.Solution}] ${
-                      op.properties?.targetResource?.resourceName
-                    } -> ${JSON.stringify(op.properties?.statusMessage, undefined, 2)}`
-                  );
-                }
-              });
-            }
+    let returnError: Error;
+    if (error.message.includes("At least one resource deployment operation failed")) {
+      let errorMessage = "resource deployments failed. ";
+      const deployments = await deployCtx.client.deployments.listByResourceGroup(
+        deployCtx.resourceGroupName
+      );
+      deployments.forEach((deployment) => {
+        if (
+          deployment.properties &&
+          deployment.properties.provisioningState === ArmDeploymentStatus.Failed &&
+          deployment.properties.error &&
+          deployment.properties.timestamp &&
+          deployment.properties.timestamp.getTime() > deployCtx.deploymentStartTime &&
+          deployment.name !== deployCtx.deploymentName
+        ) {
+          if (deployment.properties.error.details?.length) {
+            errorMessage += `(${deployment.name}) code: ${deployment.properties.error.details[0].code}, message: ${deployment.properties.error.details[0].message}.`;
+          } else {
+            errorMessage += `(${deployment.name}) code: ${deployment.properties.error.code}, message: ${deployment.properties.error.message}.`;
           }
-        });
-        pollDeploymentStatus(client, resourceGroupName, deploymentStartTime);
-      }
-    }, waitingTimeSpan);
+        }
+      });
+      returnError = new Error(errorMessage);
+    } else {
+      returnError = new Error(`Failed to deploy arm templates to azure. Error: ${error.message}`);
+    }
+    return err(returnUserError(returnError, "Solution", "ArmDeploymentFailed"));
```

Comment:
since we no longer throw an error, please also help update the telemetry part which determines the operation result based on the thrown error.

Reply:
Added it, thanks.